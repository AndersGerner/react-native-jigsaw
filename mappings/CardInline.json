[{"name":"Small Card (Inline)","tag":"CardInline","description":"An elevated card with image and a centered line of text overlayed, that takes up a third of its container.","category":"card","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Card_Inline_1col.png","supports_list_render":true,"props":{"image":{"label":"Image","description":"Image","type":"remoteImage","value":null,"editable":true},"title":{"label":"Title","description":"Text to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"textCentered":{"label":"Text centered","description":"Whether to center the text","type":"boolean","value":false,"editable":true},"elevation":{"label":"Elevation","description":"Elevation of the component. A number 0-3.","type":"flatArray","options":[0,1,2,3],"value":0,"editable":true},"aspectRatio":{"label":"Aspect ratio","description":"Aspect ratio of the image","type":"aspectRatio","value":1,"editable":true},"numColumns":{"type":"number","value":1,"editable":false}},"layout":{}},{"name":"Medium Card (Inline)","tag":"CardInline","description":"An elevated card with image and a title and description overlayed, that takes up half of its container.","category":"deprecated","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Card_Inline_2col.png","supports_list_render":true,"props":{"image":{"label":"Image","description":"Image","type":"remoteImage","value":null,"editable":true},"title":{"label":"Title","description":"Text to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"textCentered":{"label":"Text centered","description":"Whether to center the text","type":"boolean","value":false,"editable":true},"elevation":{"label":"Elevation","description":"Elevation of the component. A number 0-3.","type":"flatArray","options":[0,1,2,3],"value":0,"editable":true},"description":{"label":"Left description","description":"Text to display on the left","type":"string","value":"San Diego","editable":true},"aspectRatio":{"label":"Aspect ratio","description":"Aspect ratio of the image","type":"aspectRatio","value":1.5,"editable":true},"numColumns":{"type":"number","value":2,"editable":false}},"layout":{}},{"name":"Large Card (Inline)","tag":"CardInline","description":"An elevated card with image and a title and description overlayed, that takes up the full width of its container.","category":"card","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Card_Inline_3col.png","supports_list_render":true,"props":{"image":{"label":"Image","description":"Image","type":"remoteImage","value":null,"editable":true},"title":{"label":"Title","description":"Text to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"textCentered":{"label":"Text centered","description":"Whether to center the text","type":"boolean","value":false,"editable":true},"elevation":{"label":"Elevation","description":"Elevation of the component. A number 0-3.","type":"flatArray","options":[0,1,2,3],"value":0,"editable":true},"description":{"label":"Left description","description":"Text to display on the left","type":"string","value":"San Diego","editable":true},"aspectRatio":{"label":"Aspect ratio","description":"Aspect ratio of the image","type":"aspectRatio","value":1.5,"editable":true},"numColumns":{"type":"number","value":3,"editable":false}},"layout":{}}]