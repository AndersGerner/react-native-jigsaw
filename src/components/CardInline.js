import React from "react";
import { Image, View, Text } from "react-native";
import Card from "./Card";
import Elevation from "./Elevation";
import { withTheme } from "../core/theming";
import { COMPONENT_TYPES, FORM_TYPES } from "../core/component-types";
import Config from "./Config";
import type { Theme } from "../types";

export type CardInlineProps = {
  image: string | { uri: string },
  title?: string,
  description?: string,
  titleCentered?: boolean,
  aspectRatio?: number,
  elevation: number,
  numColumns: 1 | 2 | 3,
  theme: Theme,
  style: any,
  onPress: () => void
};

class CardInline extends React.PureComponent<CardInlineProps> {
  static defaultProps = {
    aspectRatio: 1.5,
    elevation: 2,
    numColumns: 3
  };

  render() {
    const {
      image,
      title,
      description,
      titleCentered,
      aspectRatio,
      elevation,
      numColumns,
      theme: { colors, borderRadius, typography, spacing },
      style,
      onPress
    } = this.props;

    let titleStyle, descriptionStyle;
    switch (numColumns) {
      case 1:
        titleStyle = typography.headline6;
        descriptionStyle = typography.subtitle2;
        break;
      case 2:
        titleStyle = typography.headline5;
        descriptionStyle = typography.subtitle2;
        break;
      case 3:
        titleStyle = typography.headline4;
        descriptionStyle = typography.subtitle1;
        break;
    }

    return (
      <Card style={style} onPress={onPress} numColumns={numColumns}>
        <Elevation style={{ elevation, borderRadius: borderRadius.global }}>
          <Image
            style={{
              borderRadius: borderRadius.global,
              aspectRatio
            }}
            source={typeof image === "string" ? { uri: image } : image}
            resizeMode="cover"
          />
          <View
            style={{
              position: "absolute",
              alignItems: titleCentered ? "center" : "flex-start",
              bottom: numColumns === 1 ? spacing.medium : spacing.large,
              left: spacing.large,
              right: spacing.large
            }}
          >
            {title ? (
              <Text
                numberOfLines={1}
                style={[titleStyle, { color: colors.strongInverse }]}
              >
                {title}
              </Text>
            ) : null}
            {description ? (
              <Text
                numberOfLines={1}
                style={[
                  descriptionStyle,
                  { color: colors.lightInverse, marginTop: spacing.text }
                ]}
              >
                {description}
              </Text>
            ) : null}
          </View>
        </Elevation>
      </Card>
    );
  }
}

export default withTheme(CardInline);

export const SEED_DATA = [
  {
    name: "Small Inline Card",
    tag: "CardInline1Col",
    description:
      "An elevated card with image and a centered line of text overlayed, that takes up a third of its container.",
    category: COMPONENT_TYPES.card,
    preview_image_url:
      "https://res.cloudinary.com/altos/image/upload/v1541096714/draftbit/library/jigsaw-1.0/reps/Card_Inline_1col.png",
    supports_list_render: true,
    props: {
      image: {
        label: "Image",
        description: "Image",
        type: FORM_TYPES.remoteImage,
        value: null,
        editable: true
      },
      title: {
        label: "Title",
        description: "Text to display",
        type: FORM_TYPES.string,
        value: "Beautiful West Coast Villa",
        editable: true
      },
      aspectRatio: {
        label: "Aspect ratio",
        description: "Aspect ratio of the image",
        type: FORM_TYPES.aspectRatio,
        value: 1,
        editable: true
      },
      titleCentered: {
        label: "Title centered",
        description: "Whether to center the text.",
        type: FORM_TYPES.boolean,
        value: true,
        editable: true
      },
      elevation: {
        label: "Elevation",
        description: "Elevation of the card. A number 0-3.",
        type: FORM_TYPES.number,
        value: 2,
        min: 0,
        max: 3,
        step: 1,
        precision: 0,
        editable: true
      },
      numColumns: {
        type: FORM_TYPES.number,
        value: 1,
        editable: false
      }
    },
    layout: {
      width: 375,
      height: 296
    }
  },
  {
    name: "Medium Inline Card",
    tag: "CardInline2Col",
    description:
      "An elevated card with image and a title and description overlayed, that takes up two thirds of its container.",
    category: COMPONENT_TYPES.card,
    preview_image_url:
      "https://res.cloudinary.com/altos/image/upload/v1541096650/draftbit/library/jigsaw-1.0/reps/Card_Inline_2col.png",
    supports_list_render: true,
    props: {
      image: {
        label: "Image",
        description: "Image",
        type: FORM_TYPES.remoteImage,
        value: null,
        editable: true
      },
      title: {
        label: "Title",
        description: "Text to display",
        type: FORM_TYPES.string,
        value: "Beautiful West Coast Villa",
        editable: true
      },
      description: {
        label: "Left description",
        description: "Text to display on the left",
        type: FORM_TYPES.string,
        value: "San Diego",
        editable: true
      },
      aspectRatio: {
        label: "Aspect ratio",
        description: "Aspect ratio of the image",
        type: FORM_TYPES.aspectRatio,
        value: 3 / 2,
        editable: true
      },
      titleCentered: {
        label: "Title centered",
        description: "Whether to center the text.",
        type: FORM_TYPES.boolean,
        value: false,
        editable: true
      },
      elevation: {
        label: "Elevation",
        description: "Elevation of the card. A number 0-3.",
        type: FORM_TYPES.number,
        value: 2,
        min: 0,
        max: 3,
        step: 1,
        precision: 0,
        editable: true
      },
      numColumns: {
        type: FORM_TYPES.number,
        value: 2,
        editable: false
      }
    },
    layout: {
      width: 375,
      height: 296
    }
  },
  {
    name: "Large Inline Card",
    tag: "CardInline3Col",
    description:
      "An elevated card with image and a title and description overlayed, that takes up the full width of its container.",
    category: COMPONENT_TYPES.card,
    preview_image_url:
      "https://res.cloudinary.com/altos/image/upload/v1541096652/draftbit/library/jigsaw-1.0/reps/Card_Inline_3col.png",
    supports_list_render: true,
    props: {
      image: {
        label: "Image",
        description: "Image",
        type: FORM_TYPES.remoteImage,
        value: null,
        editable: true
      },
      title: {
        label: "Title",
        description: "Text to display",
        type: FORM_TYPES.string,
        value: "Beautiful West Coast Villa",
        editable: true
      },
      description: {
        label: "Left description",
        description: "Text to display on the left",
        type: FORM_TYPES.string,
        value: "San Diego",
        editable: true
      },
      aspectRatio: {
        label: "Aspect ratio",
        description: "Aspect ratio of the image",
        type: FORM_TYPES.aspectRatio,
        value: 3 / 2,
        editable: true
      },
      titleCentered: {
        label: "Title centered",
        description: "Whether to center the text.",
        type: FORM_TYPES.boolean,
        value: false,
        editable: true
      },
      elevation: {
        label: "Elevation",
        description: "Elevation of the card. A number 0-3.",
        type: FORM_TYPES.number,
        value: 2,
        min: 0,
        max: 3,
        step: 1,
        precision: 0,
        editable: true
      },
      numColumns: {
        type: FORM_TYPES.number,
        value: 3,
        editable: false
      }
    },
    layout: {
      width: 375,
      height: 296
    }
  }
];
