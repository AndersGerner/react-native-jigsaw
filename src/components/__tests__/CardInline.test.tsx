import * as React from "react";
import renderer from "react-test-renderer";
import CardInline from "../CardInline";

it("renders card inline", () => {
  const tree = renderer
    .create(<CardInline onPress={() => {}} textCentered />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
