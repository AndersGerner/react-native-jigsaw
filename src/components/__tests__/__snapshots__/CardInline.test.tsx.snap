// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders card inline 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 718,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderColor": "rgba(18, 20, 44, 0)",
          "borderRadius": 6,
          "borderWidth": 0,
          "elevation": 2,
          "shadowColor": "rgba(18, 20, 44, 1)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.0799999982,
          "shadowRadius": 4,
          "zIndex": 100,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "aspectRatio": 1.5,
              "borderRadius": 6,
            },
          ]
        }
      >
        <Image
          resizeMode="cover"
          source={
            Object {
              "uri": "https://res.cloudinary.com/altos/image/upload/w_400,f_auto,c_scale/draftbit/components/Image.png",
            }
          }
          style={
            Array [
              Object {
                "aspectRatio": 1.5,
                "borderRadius": 6,
              },
              Object {
                "aspectRatio": undefined,
              },
              Object {
                "height": "100%",
                "width": "100%",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 16,
            "left": 16,
            "position": "absolute",
            "right": 16,
          }
        }
      />
    </View>
  </View>
</View>
`;
